src/codeu/chat/database/Database.java

```diff
@@ -0,0 +1,55 @@
+package codeu.chat.database;
+
+import java.sql.*;
+
+import codeu.chat.util.Logger;
+
+public final class Database {
+
+  private final static Logger.Log LOG = Logger.newLog(Database.class);
+
+  private String m_dbPath;
+  private Connection m_connection;
+
+  public Database(String dbPath) {
+      m_dbPath = dbPath;
+  }
+
+  /**
+   * Attempts to validate the connection.
+   * If the connection is closed or does not exist, the function will try
+   * to reconnect.
+   *
+   * @return Whether the connection was succesfully validated.
+   */
+  private boolean validateConnection() {
+    try {
+      // If the connection is null or is closed, create a new one.
+      if (m_connection == null || !m_connection.isValid(0)) {
+        Class.forName("org.sqlite.JDBC");
+        m_connection = DriverManager.getConnection(String.format("jdbc:sqlite:%s", m_dbPath));
+      }
+
+      // Any other state implies the connection is valid.
+      return true;
+    } catch (Exception ex) {
+      LOG.error("Database failed to connect: %s", ex.getMessage());
```

Comment:
When logging exceptions in java, the stack traces are very much useful. So print the entire exception object directly instead of just the message.

I think you can do `LOG.error(ex, "Database failed to connect!");`